### ISSUE TITLE
Cannot add empty headers with WinHttpOpenRequest using WINHTTP_ADDREQ_FLAG_REPLACE flag

### ISSUE THREAD

**Issue by hkleynhans** [aws/aws-sdk-cpp#471]
Cannot add empty headers with WinHttpOpenRequest using WINHTTP_ADDREQ_FLAG_REPLACE flag.  The call returns error number 12150 (ERROR_WINHTTP_HEADER_NOT_FOUND).

From the documentation of the WINHTTP_ADDREQ_FLAG_REPLACE flag (here https://msdn.microsoft.com/en-us/library/windows/desktop/aa384087(v=vs.85).aspx):

Replaces or removes a header. If the header value is empty and the header is found, it is removed. If the value is not empty, it is replaced.

So, if an empty header is passed it, the call to locate the header fails, setting the last error flag to 12150.

Should this flag be dropped?

**Comment by JonathanHenson** [comment]
Are you using the HttpClient directly or is this going through one of our SDKs?

**Comment by hkleynhans** [comment]
I ran into the issue while adding AuthSig V2 support.  It requires the Content-Type and Content-MD5 as part of the hashed string.  I ported the Java code which adds empty headers in the event that those required have not been populated.  The same code which works fine under Linux & Mac using Curl does not work with Windows.

I will also try solve this another way - by removing the empty headers before it is passed to the HTTP implementation.
